nixos/akkoma: Don't warn if no installWrapper

kivikakk opened the pull request:
## Description of changes

Currently, `nixos/akkoma` issue a warning relating to its `installWrapper` option if `sudo` (and `sudo-rs`) aren't installed.  This is fine, except we get the warning even if we explicitly turn off `installWrapper` -- say, for this very reason!

Rather than warning on every build until either sudo is installed or Akkoma is uninstalled, only warn if `cfg.installWrapper` is true.

## Things done

- Built on platform(s)
  - [x] x86_64-linux — tested that the warning is no longer emitted when `config.services.akkoma.installWrapper = false;`, and still emitted when unset/set to true explicitly.
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

tomfitzhenry reviewed (approved):
LGTM.

Sorry, but through no fault of your own, this PR no longer applies to `master`. Happy to rebase this?

tomfitzhenry reviewed (changes requested):
Marking as "Request changes".

kivikakk commented:
Sure thing, on it!

kivikakk committed:
nixos/akkoma: Don't warn if no installWrapper

Currently the installWrapper warning is issued if sudo (and sudo-rs)
aren't installed.  This is fine, except we get the warning even if we
explicitly turn off installWrapper -- say, for this very reason!

Rather than warning on every build until either sudo is installed or
Akkoma is uninstalled, only warn if cfg.installWrapper is true.

tomfitzhenry reviewed (approved):
LGTM, nice change, sorry it took so long to get merged!
